pkgs/build-support/dotnet/build-dotnet-module/fetchDeps.sh

```diff
@@ -0,0 +1,110 @@
+@shell@
```

Comment:
Minor nitpick, but it seems to be much more common to do something like `#! @shell@`.  I think I'd prefer it that way because currently it's not actually the shell path, it's a shebang containing the shell path.

Reply:
I'd strongly prefer not to do that, because then there's no way to prevent Shellcheck's integration into my editor from complaining.

Reply:
So I assume you get those errors on e.g. `nixos/modules/system/boot/stage-1-init.sh`?

I'm not sure I like the idea of going against the grain like this.  Couldn't any of the template substitutions confuse the checker?

Reply:
Yep, stage-1-init.sh contains a whole bunch of Shellcheck errors, the first of which is the invalid hashbang. (Most of those errors look legit: unquoted interpolations, `cd`s that can silently fail, unused variables, `read` which mangles backslashes, etc.)

I've (hopefully) carefully phrased the substitutions so that they are all "valid Bash" both before and after substitution. I consider it a bug in this PR if any of them are invalid. (There are nine Shellcheck errors in fetch-deps.sh, all of which would have been errors before this PR except that it was all hidden within a Nix string; none of the errors are related to the substitutions.)

Reply:
I could certainly go further: in one of my projects somewhere, I've written a `makeShellScript` of some kind which does substitutions and then runs Shellcheck against the result post-substitution. That would be the gold standard, although fixing errors under that workflow *with the shell script template as an inline Nix string* would still be more annoying vs fixing errors under that workflow *with the shell script template in a separate .sh file*.

Reply:
```
In [...]/nixos/modules/system/boot/stage-1-init.sh line 1:
#! @shell@
^-- SC1008 (error): This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh/'busybox sh'. Add a 'shell' directive to specify.
^--------^ SC2239 (error): Ensure the shebang uses an absolute path to the interpreter.
```

I think you just need to add `# shellcheck shell=bash` on the next line to suppress it.  This will likely also help it check more accurately than if it couldn't find a shebang at all.

Reply:
Oh that does work! Sorry, I thought those directives had to go right at the top of the file, but of course on second thoughts that's a silly idea (how could that possibly work with any existing hashbangs). I'll push that up now.

Reply:
Done, and I've updated the diff in the original description.